File: Advantage Shuffle/Suggested.java

import java.util.Arrays;

public class Suggested {

    public static void main(String[] args) {
        Solution a = new Solution();
        System.out.println(Arrays.toString(a.advantageCount(new int[]{1, 2, 3, 4, 5}, new int[]{3, 5, 4, 6, 2})));
        System.out.println(Arrays.toString(a.advantageCount(new int[]{1, 4, 2, 1, 3}, new int[]{2, 3, 1, 2, 2})));
        System.out.println(Arrays.toString(a.advantageCount(new int[]{2, 7, 11, 15}, new int[]{1, 10, 4, 11})));
        System.out.println(Arrays.toString(a.advantageCount(new int[]{12, 24, 8, 32}, new int[]{13, 25, 32, 11})));
    }
}

class Solution {

    public int[] advantageCount(int[] nums1, int[] nums2) {
        Integer[] ord = new Integer[nums2.length];
        int[] ans = new int[nums1.length];
        for (int i = 0; i < nums2.length; i++) {
            ord[i] = i;
        }
        Arrays.sort(ord, (a, b) -> Integer.compare(nums2[a], nums2[b]));
        Arrays.sort(nums1);
        int i = 0, j = nums2.length - 1;
        for (int index : ord) {
            if (nums1[j] > nums2[index]) {
                ans[index] = nums1[j];
                j--;
            } else {
                ans[index] = nums1[i];
                i++;
            }
        }
        return ans;
    }
}

File: Advantage Shuffle/Suggested.py
class Solution:
    def advantageCount(self, nums1, nums2):
        length = len(nums1)
        result = [-1] * length
        nums1.sort(reverse=True)
        indexes = [i for i in range(length)]
        indexes.sort(key=lambda i: nums2[i],reverse=True)
        left, right = 0, length -1
        for index in indexes:
            if nums1[left] > nums2[index]:
                result[index] = nums1[left]
                left+=1
            else:
                result[index] = nums1[right]
                right-=1
        return result


if __name__ == "__main__":
    a = Solution()
    print(a.advantageCount([1, 2, 3, 4, 5], [3, 5, 4, 6, 2]))
    print(a.advantageCount([1, 4, 2, 1, 3], [2, 3, 1, 2, 2]))
    print(a.advantageCount([2, 7, 11, 15], [1, 10, 4, 11]))
    print(a.advantageCount([12, 24, 8, 32], [13, 25, 32, 11]))
